ksql-common/src/main/java/io/confluent/ksql/util/KsqlServerException.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License; you may not use this file
+ * except in compliance with the License.  You may obtain a copy of the License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.util;
+
+import org.apache.kafka.streams.errors.StreamsException;
+
+/**
+ * A typed error which represents an internal server issue as opposed to
+ * {@link KsqlException}, which represents a user error.
+ */
+public class KsqlServerException extends StreamsException {
```

Comment:
Review note: I created this class as a way to differentiate user errors (e.g. bad statements) from server errors (e.g. could not communicate with Kafka). For now, I only use it in one place but going forward we should differentiate the two types of errors for easier error handling.